plugin.rb

```diff
@@ -186,11 +199,10 @@ class ::TopicListItemSerializer
   require_dependency 'list_controller'
   class ::ListController
     generate_message_route(:private_messages_assigned)
-    generate_message_route(:messages_assigned)
   end
 
   add_to_class(:topic_query, :list_messages_assigned) do |user|
-    list = joined_topic_user.where("
+    list = default_results.where("
```

Comment:
(I know I asked you to use `default_results` originally, sorry 😅 )

I realised that `default_results` doesn't include PMs. We like to have a combined list of 'topics and PMs' for the assigned list. So using `joined_topic_user` worked well for that in the past. 

However... now that we are allowing non-admins to see this list, we need to add some security. I think something like this could work (and also the same on the `group_messages_assigned` method) 

Please test this though - I haven't checked it

```suggestion
    secure = Topic.listable_topics.secured(@guardian).or(Topic.private_messages_for_user(@user))
    list = joined_topic_user.where(secure).where("
```